### PR TITLE
build: update to rules_nodejs 0.38

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

**Comment by alexeagle** [comment]
Caretaker: cl/271704225 needs to be patched in along with this change, so that the `.mjs` extensions remain `.closure.js` for google3

**Comment by gregmagolan** [comment]
@googlebot I consent

File: packages/bazel/src/builders/files/src/BUILD.bazel.template [hunk 1]
```diff
@@ -2,7 +2,7 @@ package(default_visibility = ["//visibility:public"])
 
 load("@npm_angular_bazel//:index.bzl", "ng_module")
 load("@npm_bazel_karma//:index.bzl", "ts_web_test_suite")
-load("@build_bazel_rules_nodejs//:defs.bzl", "rollup_bundle", "history_server")
+load("@build_bazel_rules_nodejs//:index.bzl", "rollup_bundle", "history_server")
```
Comment (gregmagolan) [inline packages/bazel/src/builders/files/src/BUILD.bazel.template]: Overlooked `history_server` here?

Comment (gregmagolan) [reply packages/bazel/src/builders/files/src/BUILD.bazel.template]: Done

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by alexeagle** [comment]
CARETAKER: see note above about g3 CL needed to patch in with this

**Review by JiaLiPassion — APPROVED** [review]

**Comment by alxhub** [comment]
Note: I'm merging this to `master` only as it does not apply cleanly on the patch branch. @alexeagle will open a separate PR for patch-only.

**Pull request closed without merging**